Userland/Libraries/LibWeb/CSS/StyleProperties.cpp

```diff
@@ -325,6 +325,20 @@ Optional<CSS::StrokeLinecap> StyleProperties::stroke_linecap() const
     return keyword_to_stroke_linecap(value->to_keyword());
 }
 
+Optional<CSS::StrokeLinejoin> StyleProperties::stroke_linejoin() const
+{
+    auto value = property(CSS::PropertyID::StrokeLinejoin);
+    return keyword_to_stroke_linejoin(value->to_keyword());
+}
+
+Optional<float> StyleProperties::stroke_miterlimit() const
+{
+    auto value = property(CSS::PropertyID::StrokeMiterlimit);
+    if (!value->is_number())
+        return {};
+    return value->as_number().number();
+}
```

Comment:
1. As mentioned, this doesn't need to return `Optional`.
2. It's a number, so it can also be a math value, which might not be resolvable until later. So this wants to return `NumberOrCalculated`. Similarly ComputedValues's `stroke_miterlimit` wants to be that.

Reply:
Thanks! I modeled this after `flex_grow` – I suppose that has the same problem then?

Reply:
Indeed. I think really we just want to replace a lot of the property-specific getters with general ones like `StyleProperties::length_percentage(PropertyID)`, but maybe it's clearer to have named getters rather than having to know in advance what type you want? IDK.

Reply:
Also, since this is just a number, not a length, what's a situation where evaluation has to wait until later? `StyleProperties::resolve_opacity_value()` seems to eagerly evaluate – are there problems with that approach?

Reply:
Yeah... and there are larger issues around how we handle non-lengths in calculations:

Since we originally implemented `calc()`, the spec changed to allow crazier type arithmetic. So you can do things like `calc(1em / 1rem)` and it'll evaluate to a `<number>`, but that of course can't be resolved until we have the font sizes calculated, and we'd also need to pass in a length context. (None of the `resolve_thing_that_isn't_length()` methods let you pass in that context yet, and we'll probably want to do things a smarter way to avoid generating that context when most of the time it's not needed.)

Reply:
> Also, not 100% sure about the `VERIFY(math_value.resolves_to_number());` – but Properties.json might guarantee that, and nothing seems to explode if I try to set `calc(2px)` or `calc(10%)` locally, so it _seems_ fine (?)

Oops, I missed this message somehow. Yeah that should be fine.